django/db/models/sql/query.py

```diff
@@ -2338,6 +2345,34 @@ def add_ordering(self, *ordering):
         else:
             self.default_ordering = False
 
+    @property
+    def orderby_issubset_groupby(self):
+        """
+        Return true if order_by of query is subset of group_by.
+        """
+        # we don't want to harm original query, so we need to clone it
+        q = self.clone()
+        if q.group_by is False:
+            # there is no use case for that, but we need to be sure
+            return False
+        if q.group_by in (None, True):
+            # it seems like there is no aggregation at all (None)
+            # or there are all required groupbies(True) generated automatically
+            # from models fields - so, it is safe to clear ordering
+            return True
+        order_by_set = set(
+            [
+                (
+                    order_by.resolve_expression(q)
+                    if hasattr(order_by, "resolve_expression")
+                    else F(order_by).resolve_expression(q)
+                )
+                for order_by in q.order_by
+            ]
+        ).union(q.extra_order_by)
+        group_by_set = set([group_by.resolve_expression(q) for group_by in q.group_by])
+        return order_by_set.issubset(group_by_set)
```

Comment:
Do we have members of `group_by` which are not already resolved? I thought they must be already resolved since they are plucked from `Query.select` and `annotations`.

https://github.com/django/django/blob/020e5799ad74cfafd469f032cd05767c9d670a16/django/db/models/sql/query.py#L2383-L2391

assuming this is the case it means it could be reduced to

```suggestion
        return order_by_set.issubset(self.group_by)
```